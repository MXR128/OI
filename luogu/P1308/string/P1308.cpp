#include <iostream>
#include <string>
#include <cstdio>

using namespace std;

string key, sen, cmp;
int l=-1,r;
int count = 0, first = -1;

int main() {
	freopen("testdata.in", "r", stdin);
	cin >> key; getchar();
	getline(cin, sen);
	for (int i = 0; i < (int)key.size(); i++)
		if (key[i] >= 'A' && key[i] <= 'Z')
			key[i] += 32;
	for (int i = 0; i < (int)sen.size(); i++)
		if (sen[i] >= 'A' && sen[i] <= 'Z')
			sen[i] += 32;
	while(1==1){
		r=sen.find(' ',l+1);
		if(sen[r]!=' ')break;
		cmp.assign(sen,l+1,r-l-1);
		if(cmp==key){
			count++;
			if(first==-1)first=l+1;
			// cout<<l<<' '<<r<<endl;
		}
		l=r;
	}
	if (first == -1) {
		cout << -1;
	} else {
		cout << count << ' ' << first;
	}
	return 0;
}