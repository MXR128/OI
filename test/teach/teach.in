1
3.1415926
A
Accept