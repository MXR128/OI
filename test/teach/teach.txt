//teach

/**
 * 注意事项:
 * 题目应该都会是很简单的模拟题,所以
 * 		审题!审题!审题!
 * 重要的事情说三遍
 * 语法错误编译时会报错,仔细检查/回忆
 * DevC++下 编译F9 运行F10 编译并运行F11
 */

//必备头文件
#include <stdio.h>

//常用头文件
#include <stdlib.h>
#include <string.h>

int cmpS(const void *m,const void *n);

int main(){

//输入输出模块

	/*
		文件读写
		freopen("文件名","使用方式",端口);
		文件名用
	 */
	freopen("teach.in","r",stdin);
	freopen("teach.out","w",stdout);

	/*
		变量类型
			整数int 小数double 字符char
		变量定义
			变量类型 变量名(=赋初值)
	 */
	int _int=12;
	double _double=3.1415926;
	char _char='A';
	/*
		数组定义
			变量类型 变量名[一维大小][二维大小]... (={赋初值} 几种方式 全部赋值为0:={0} 逐个赋值:={1,2,3...} 多维数组逐个赋值:={{1,2},{3,4}})
		数组名就是这个数组的地址,字符串同
	 */
	double a[2],b[3]={0},c[4]={1,2,3,4},d[2][2]={{1,2},{3,4}};
	/*
		字符数组(字符串)
			定义同上
			字符串表示为"字符串",字符表示为'单个字符'
			字符串以'\0'结尾.输入时会自动加
			字符/字符串以ASCII存储字符,里面0-9,a-z,A-Z都是连续的
	 */
	char _string[300]="Accept";

	/*
		运算
			运算符号 加法+ 减法- 乘法* 除法/ 取余(mod,模运算)整数%整数 三目运算符 条件?值1:值2 位移运算 << >>
			取余运算必须是整数的运算
			整数之间的除法是去尾的整除
			整数和小数的运算结果是小数
			三目运算符的条件为真时表达式的值为值1,否则为值2
			位移运算 
				左移<< 右移>>
				将数字在二进制中每位向左或向右移n位
				所以若有变量a,移动n位:(a^b表示a的b次方)
					a<<n 与 a*(2^n) 等价 
					a>>n 与 a*(2^-n) 等价
		赋值
			被赋值变量=值;
			因为是从右向左赋值,可以
				变量1=变量2=变量3;
			特殊赋值符号 运算= 如+= -= *= /= %= <<= >>=(不适用于三目运算符)
			变量+=(值) 与 变量=变量+(值) 等价 
		自加/自减
			变量++,++变量 与 变量+=1 几乎等价,区别在于:
				变量++ 的值是 变量
				++变量 的值是 变量+1
		另外想起一点挺坑的, 0/-1=-0,输出的时候会多个负号
	 */
	
	/*
		一般输入输出
		输入/输出一个字符串,中间内容可被格式符替换
	*/
	scanf("%d %lf %c %s",&_int,&_double,&_char,_string);
	//记得变量要加取地址符&,字符数组不用
	printf("%d %lf %c %s\n",_int,_double,_char,_string);
	//换行符'\n' ,字符'%'为'%%',字符'\'为 '\\'

	//特殊格式输出
	printf("%2d %-2d %.2lf",_int,_int,_double);
	//两单位右对齐/两单位左对齐
	//保留两位小数

// 语法模块
	//if for while中的 语句; 可换为{语句1;语句2;...}

	/*
		条件运算
		比较两个值的 等于== 不等于!= 大于> 小于<
		两个条件的关系 与&& 或||
		取反一个条件 !(条件)
	 */

	/*
	if
		if(条件)
			语句
		else
			语句
	 */
	if(1==1){

	}else{

	}

	// 循环

	/*
	for
		定义循环变量
		for(循环变量初始化;循环条件;循环后操作)
			语句
		我已经试过了,你考试用的版本DevC++5.6的C语言for语句内不能定义循环变量
	 */
	int i;
	for(i=1;i<=10;i++){
		//Code
	}

	/*
	while
		while(循环条件)
			语句 

		do
			语句
		while(循环条件);

		do-while语句后面要加分号,因为整个do-while视为一个语句
	 */
	
	while(i>0)
		--i;

	do
		++i;
	while(i<=10);

	// 注意思考循环条件,避免死循环

//其他函数

	//sizeof(变量/数组名) 返回变量或数组的大小,单位是字节

	/*
		数组的按字节赋值
		这个函数被包含在 string.h 里面
		memset(地址,大小,数值);
		注意:一个int占4个字节,所以需要赋值为准确数字的时候不能按字节赋值为除0,-1以外的数
			0在二进制中是0000 0000,-1是 1111 1111所以可以
			如果需要赋值为一个大数,可以用0x3f
	 */
	memset(_string,sizeof(_string),0x3f);

	/*
		数组的排序
		包含在 stdlib.h 中
		qsort(数组地址,元素个数,单个元素的大小,比大小函数);
		这里要写一个比大小函数,我写成cmpS
	 */

	qsort(b,3,sizeof(b[0]),cmpS);

	return 0;
}

/*
	比大小函数的返回类型和传参类型是固定的,函数名和对比方式可以改变,根据题意
	取内容符*
	*加地址 得到的是这个地址里存的值
	因为传入的是void类型的指针,所以要加一个指针类型的定义 (变量类型 *)变量
 */
int cmpS(const void *m,const void *n){
	if(*(double *)m==*(double *)n)
		return 0;
	return *(double *)m < *(double *)n;
}
