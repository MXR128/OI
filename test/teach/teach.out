1 3.141593 A Accept
 1 1  3.14